### PR TITLE
Fixes:#5795:MSI support for slots

### PR DESCRIPTION
**Pull request by panchagnula**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5796
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -241,25 +241,28 @@ def _list_app(cli_ctx, resource_group_name=None):
...
     client = web_client_factory(cmd.cli_ctx)
 
     def getter():
-        return _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'get')
+        return _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'get', slot)
 
     def setter(webapp):
         webapp.identity = ManagedServiceIdentity(type='SystemAssigned')
-        poller = client.web_apps.create_or_update(resource_group_name, name, webapp)
+        if slot is None:
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py]: can you also use `_generic_site_operation` here?

**Comment by yugangw-msft** [comment]
@panchagnula, the CI error is a known issue, so don't worry about it. One thing to confirm though, that you did verify it works , and this flag can be propagated to other slots through swap. If yes, let me know and I will merge

**Comment by panchagnula** [comment]
@yugangw-msft I did validate the slots functionality with the change. Thanks for the info on the CLI error.

**Pull request merged**